.github/workflows/continuous-integration.yml

```diff
@@ -45,21 +45,42 @@ jobs:
 
       - name: List changed files
         id: changed-files
+        if: ${{ github.ref != 'refs/heads/master' }}
         run: echo ::set-output name=all_changed_files::$(git diff --name-only origin/master...)
 
+      - name: List changed files - master
+        id: changed-files-master
+        if: ${{ github.ref == 'refs/heads/master' }}
+        run: echo ::set-output name=all_changed_files::$(git diff --name-only HEAD~)
```

Comment:
Is it possible to combine this into a composite action with the other `List changed files` step, and have the action accept an `is_master` input? That way, we'd only need one `changed-files` step. Or maybe we can have one `changed-files` step and handle the logic in bash rather than GHA.

Having a single step for getting the changed files would also prevent us from needing two `get-changed-apps` steps and two `Build` steps